os/fs/mount/fs_mount.c

```diff
@@ -331,17 +332,42 @@ int mount(FAR const char *source, FAR const char *target, FAR const char *filesy
 		 * error.
 		 */
 
-		fdbg("ERROR: Bind method failed: %d\n", ret);
+		if (ret == -ENODEV && strncmp(filesystemtype, "smartfs", 7) == 0) {
```

Comment:
all specific block driver or file system or character device hided in vfs properly, you should not put specific code here, don't break our design principle of layered architecture

Reply:
The mtd driver layer only accesses the block driver instance or the mountpoint instance.
Since the block device path is not known to the fs/mtd driver layer, we cannot reformat the partition from that scope.